cypress/integration/tests/controls/stakeholders/pagination.test.ts

```diff
@@ -0,0 +1,213 @@
+/// <reference types="cypress" />
+
+import { login, clickByText, selectItemsPerPage, click } from "../../../../utils/utils";
+import { navMenu, navTab } from "../../../views/menu.view";
+import { controls, stakeholders, tdTag, trTag } from "../../../types/constants";
+
+import { Stakeholders } from "../../../models/stakeholders";
+
+import * as data from "../../../../utils/data_utils";
+import {
+    firstPageButton,
+    lastPageButton,
+    nextPageButton,
+    pageNumInput,
+    prevPageButton,
+    appTable,
+    deleteButton,
+    confirmButton,
+} from "../../../views/common.view";
+
+var stakeholdersList: Array<Stakeholders> = [];
+
+describe("Stakeholder pagination validations", function () {
+    before("Login and Create Test Data", function () {
+        // Perform login
+        login();
+
+        // Navigate to stakeholder tab
+        clickByText(navMenu, controls);
+        clickByText(navTab, stakeholders);
+        var rowsToCreate = 0;
+
+        // Get the current table row count and create appropriate test data rows
+        selectItemsPerPage(100);
+        cy.wait(2000);
+        cy.get(appTable)
+            .next()
+            .then(($div) => {
+                if (!$div.hasClass("pf-c-empty-state")) {
+                    cy.get("td[data-label=Email]").then(($rows) => {
+                        var rowCount = $rows.length;
+                        if (rowCount <= 10) {
+                            rowsToCreate = 11 - rowCount;
+                        }
+                        if (rowsToCreate > 0) {
+                            // Create multiple stakeholders
+                            for (let i = 0; i < rowsToCreate; i++) {
+                                const stakeholder = new Stakeholders(
+                                    data.getEmail(),
+                                    data.getFullName()
+                                );
+                                stakeholder.create();
+                                stakeholdersList.push(stakeholder);
+                            }
+                        }
+                    });
+                } else {
+                    rowsToCreate = 11;
+                    // Create multiple stakeholders
+                    for (let i = 0; i < rowsToCreate; i++) {
+                        const stakeholder = new Stakeholders(data.getEmail(), data.getFullName());
+                        stakeholder.create();
+                        stakeholdersList.push(stakeholder);
+                    }
+                }
+            });
+    });
```

Comment:
This can be done as :

     rowsTotest = 11;
     cy.get(appTable)
            .next()
            .then(($div) => {
                if (!$div.hasClass("pf-c-empty-state")) {
                    cy.get("td[data-label=Email]").then(($rows) => {
                        var rowCount = $rows.length;
                    }
            if (rowCount > rowstotest):
                  do nothing ;
           else : 
                  rowsToCreate = rowsTotest - rowCount
            for (let i = 0; i < rowsToCreate; i++) {
                        const stakeholder = new Stakeholders(data.getEmail(), data.getFullName());
                        stakeholder.create();
                        stakeholdersList.push(stakeholder);
                    }

Reply:
I tried it this way to use the loop only once. But value of var rowCount is not accessible outside if-else, since the scope of Promise gets over there itself.